doc/source/tune/_tutorials/tune-sklearn.py

```diff
@@ -0,0 +1,147 @@
+
+Run ``pip install ray[tune] tune-sklearn`` to get started.
+
+Walkthrough
+-----------
+
+``tune-sklearn`` has two APIs: TuneSearchCV, and TuneGridSearchCV. They are drop-in replacements for Scikit-learn's RandomizedSearchCV and GridSearchCV.
+
+For this example, let’s start by using ``TuneGridSearchCV`` with sklearn’s `digits dataset`_ and a `SGDClassifier`_ to classify digits.
+
+.. _`digits dataset`: https://scikit-learn.org/stable/modules/generated/sklearn.datasets.load_digits.html
+.. _`SGDClassifier`: https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html
+
+To start out, change the import statement to get tune-scikit-learn’s grid search cross validation interface:
+
+"""
+# from sklearn.model_selection import GridSearchCV
+from tune_sklearn import TuneGridSearchCV
+
+#######################################################################
+# And from there, we would proceed just like how we would in Scikit-Learn’s interface!
+#
+# The `SGDClassifier`_ has a ``partial_fit`` API, which enables it to stop fitting to the data for a certain hyperparameter configuration. If the estimator does not support early stopping, we would fall back to a parallel grid search.
+
+# Other imports
+from sklearn.model_selection import train_test_split
+from sklearn.linear_model import SGDClassifier
+from sklearn import datasets
+import numpy as np
+
+# Load in data
+digits = datasets.load_digits()
+
+# Set training and test sets
+x = digits.data
+y = digits.target
+x_train, x_test, y_train, y_test = train_test_split(x, y, test_size=.2)
+
+# Example parameters to tune from SGDClassifier
+parameter_grid = {"alpha": [1e-4, 1e-1, 1], "epsilon": [0.01, 0.1]}
+
+#######################################################################
+# As you can see, the setup here is exactly how you would do it for Scikit-Learn. Now, let's try fitting a model.
+
+tune_search = TuneGridSearchCV(
+    SGDClassifier(),
+    parameter_grid,
+    scheduler="MedianStoppingRule",
+    early_stopping=True,
+    max_iters=10)
+
+import time  # Just to compare fit times
+start = time.time()
+tune_search.fit(x_train, y_train)
+end = time.time()
+print("Tune Fit Time:", end - start)
+
+#######################################################################
+# Note the slight differences we introduced above:
+#
+#  * a `scheduler`, and
+#  * a specification of `max_iters` parameter
+#
+# The ``scheduler`` determines when to stop early - MedianStoppingRule is a great default, but see :ref:`Tune's documentation on schedulers <tune-schedulers>` here for a full list to choose from. ``max_iters`` is the maximum number of iterations a given hyperparameter set could run for; it may run for fewer iterations if it is early stopped.
+#
+# Try running this compared to the GridSearchCV equivalent, and see the speedup for yourself!
```

Comment:
It may not actually be faster because this dataset is too small

Reply:
Would be resolved if we use `make_classification` for slightly larger datasets

Reply:
done